src/net_processing.cpp

```diff
@@ -2878,20 +2878,20 @@ void PeerManagerImpl::ProcessHeadersMessage(CNode& pfrom, Peer& peer,
 /**
  * Reconsider orphan transactions after a parent has been accepted to the mempool.
  *
- * @param[in,out]  orphan_work_set  The set of orphan transactions to reconsider. Generally only one
+ * @peer[in]  peer                  The peer whose orphan transactions we will reconsider. Generally only one
  *                                  orphan will be reconsidered on each call of this function. This set
  *                                  may be added to if accepting an orphan causes its children to be
  *                                  reconsidered.
  */
-void PeerManagerImpl::ProcessOrphanTx(std::set<uint256>& orphan_work_set)
+void PeerManagerImpl::ProcessOrphanTx(Peer& peer)
```

Comment:
After the work set is moved to the orphanage, `ProcessOrphanTx` really only needs the id of the peer not the entire `Peer`.

Maybe pass `NodeId` to `ProcessOrphanTx` instead of `Peer` in 9910ed755c3dfd7669707b44d993a20030dd7477 and have it look up the needed `Peer`.

Reply:
If you already have a `Peer`, I don't think there's really any reason not to pass in.